src/robotFactory.js

```diff
@@ -1,10 +1,100 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(name, weight, coords, chipVersion) {
+    this.name = name;
+    this.weight = weight;
 
-class FlyingRobot {}
+    this.coords = {
+      x: coords.x || 0,
+      y: coords.y || 0,
+    };
 
-class DeliveryDrone {}
+    this.chipVersion = chipVersion;
+  }
+
+  goForward(step = 1) {
+    this.coords.y += step;
+
+    return this;
+  }
+
+  goBack(step = 1) {
+    this.coords.y -= step;
+
+    return this;
+  }
+
+  goLeft(step = 1) {
+    this.coords.x -= step;
+
+    return this;
+  }
+
+  goRight(step = 1) {
+    this.coords.x += step;
+
+    return this;
+  }
+
+  getInfo() {
+    return `Robot: ${this.name}, Chip version: ${this.chipVersion}, `
+      + `Weight: ${this.weight}`;
+  }
+}
+
+class FlyingRobot extends BaseRobot {
+  constructor(name, weight, coords, chipVersion) {
+    super(name, weight, coords, chipVersion);
+
+    this.coords = {
+      x: coords.x || 0,
+      y: coords.y || 0,
+      z: coords.z || 0,
+    };
+  }
+
+  goUp(step = 1) {
+    this.coords.z += step;
+
+    return this;
+  }
+
+  goDown(step = 1) {
+    this.coords.z -= step;
+
+    return this;
+  }
+}
+
+class DeliveryDrone extends FlyingRobot {
+  constructor(
+    name,
+    weight,
+    coords,
+    chipVersion,
+    maxLoadWeight,
+    currentLoad
+  ) {
+    super(name, weight, coords, chipVersion);
+    this.maxLoadWeight = maxLoadWeight;
+    this.currentLoad = currentLoad || null;
```

Comment:
```suggestion
    this.currentLoad = currentLoad;
```
maybe you are right, but  to my mind it is unnecessary